pkg/apis/capabilities/v1beta1/openapi_types.go

```diff
@@ -0,0 +1,193 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	OpenAPIKind = "OpenAPI"
+
+	// OpenAPIInvalidConditionType represents that the combination of configuration
+	// in the OpenAPISpec is not supported. This is not a transient error, but
+	// indicates a state that must be fixed before progress can be made.
+	// Example: the spec references invalid openapi spec
+	OpenAPIInvalidConditionType common.ConditionType = "Invalid"
+
+	// OpenAPIReadyConditionType indicates the openapi resource has been successfully reconciled.
+	// Steady state
+	OpenAPIReadyConditionType common.ConditionType = "Ready"
+
+	// OpenAPIFailedConditionType indicates that an error occurred during reconcilliation.
+	// The operator will retry.
+	OpenAPIFailedConditionType common.ConditionType = "Failed"
+)
+
+// OpenAPIRefSpec Reference to the OpenAPI Specification
+type OpenAPIRefSpec struct {
+	// ConfigMapRef ConfigMap that contains the OpenAPI Document
+	// +optional
+	ConfigMapRef *corev1.ObjectReference `json:"configMapRef,omitempty"`
+
+	// URL Remote URL from where to fetch the OpenAPI Document
+	// +kubebuilder:validation:Pattern=`^https?:\/\/.*$`
+	// +optional
+	URL *string `json:"url,omitempty"`
+}
+
+// OpenAPISpec defines the desired state of OpenAPI
+type OpenAPISpec struct {
+	// OpenAPIRef Reference to the OpenAPI Specification
+	OpenAPIRef OpenAPIRefSpec `json:"openapiRef"`
+
+	// ProviderAccountRef references account provider credentials
+	// +optional
+	ProviderAccountRef *corev1.LocalObjectReference `json:"providerAccountRef,omitempty"`
+
+	// ProductionPublicBaseURL Custom public production URL
+	// +kubebuilder:validation:Pattern=`^https?:\/\/.*$`
+	// +optional
+	ProductionPublicBaseURL string `json:"productionPublicBaseURL,omitempty"`
+
+	// StagingPublicBaseURL Custom public staging URL
+	// +kubebuilder:validation:Pattern=`^https?:\/\/.*$`
+	// +optional
+	StagingPublicBaseURL string `json:"stagingPublicBaseURL,omitempty"`
+
+	// ProductSystemName 3scale product system name
+	// +optional
+	ProductSystemName string `json:"productSystemName,omitempty"`
+
+	// PrivateBaseURL Custom private base URL
+	// +optional
+	PrivateBaseURL string `json:"privateBaseURL,omitempty"`
+
+	// PrefixMatching Use prefix matching instead of strict matching on mapping rules derived from openapi operations
+	// +optional
+	PrefixMatching bool `json:"prefixMatching,omitempty"`
+
+	// PrivateAPIHostHeader Custom host header sent by the API gateway to the private API
+	// +optional
+	PrivateAPIHostHeader string `json:"privateAPIHostHeader,omitempty"`
+
+	// PrivateAPISecretToken Custom secret token sent by the API gateway to the private API
+	// +optional
+	PrivateAPISecretToken string `json:"privateAPISecretToken,omitempty"`
```

Comment:
Shouldn't this be a secret reference?

Reply:
This is a 3scale token, so called "Secret Token". Nothing to do with k8s secrets
![Screenshot_2020-10-29 Services - Settings Red Hat 3scale API Management](https://user-images.githubusercontent.com/881529/97592744-4895d300-1a01-11eb-9444-716b03c3c05c.png)

Reply:
But is that token sensitive information? If so, it should be in a secret.

Reply:
ok, let it be a secret.

Reply:
@miguelsorianod this field exists in `products.capabilities.3scale.net`. So it should also be updated to a secret. That would break backwards compat. I see your concern, but the requested change does not belong to this PR. 

I have opened issue for that: https://github.com/3scale/3scale-operator/issues/507

Reply:
Changing it in here would not break backwards compatibility no?
It would break backwards compatibility if we were to change it in `Product`, but not here.
We should be able to send it to `Product` as it expects without breaking compatibility

Reply:
Does not make sense having the info in a secret for the openapi CR and open field in the product CR. This issue has to be tacked separately.

Reply:
I don't see why it does not make sense. They are different objects with different interfaces.
If we leave this to be changed and the issue for some reason were to be tackled after a release then it would be even harder.

Up to you but I don't see the need for this to be tackled separated for this resource.

Reply:
It does not make sense because the purpose of the keeping some sensitive data in a secret does not apply. The token will be visible in the Product CR. I would prefer having now as it is and fix the issue separately for both OpenAPI CRD and Product CRD, making both reference to a secret. So the sensitive data lives in a single place "securized" in a secret.

Reply:
:+1: as you prefer